dot/metrics/collector.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import (
+	"time"
+
+	ethmetrics "github.com/ethereum/go-ethereum/metrics"
+)
+
+// GaugeCollector abstracts the Update function to collectors
+// just implement the collection
+type GaugeCollector interface {
```

Comment:
```suggestion
type GaugeMetric interface {
type GaugeMetricSource interface {
type GaugeMetricProvider interface {
```
`GaugeCollector` sounds like the one that collects over all these sources/providers/metrics.